lib/server/routes/buckets.js

```diff
@@ -150,30 +150,77 @@ BucketsRouter.prototype.createBucket = function(req, res, next) {
 BucketsRouter.prototype.destroyBucketById = function(req, res, next) {
   const Bucket = this.storage.models.Bucket;
   const BucketEntry = this.storage.models.BucketEntry;
+  const StorageEvent = this.storage.models.StorageEvent;
 
   log.info('removing bucket %s for %s', req.params.id, req.user._id);
 
-  Bucket.findOne({
-    _id: req.params.id,
-    user: req.user._id
-  }, function(err, bucket) {
-    if (err) {
-      return next(new errors.InternalError(err.message));
+  BucketEntry.aggregate([
+    {
+      $match: {
+        bucket: req.params.id
+      }
+    },
+    {
+      $lookup: {
+        from: 'frames',
+        localField: 'frame',
+        foreignField: '_id',
+        as: 'frameData'
     }
+    },
+    {
+      $unwind: {
+        path: '$frameData'
+      }
+    },
+    {
+      $project: {
+        _id: 0,
+        bucket: '$bucket',
+        bucketEntry: '$_id',
+        user: {$literal: req.user._id},
+        downloadBandwidth: {$literal: 0},
+        storage: {$multiply: [-1, '$frameData.size']}
+      }
+    }], 
+    function(err, results) {
+      if (err) {
+        log.warn('destroyBucketById: storage event aggregation failed, reason: %s',
+                err.message);
+      }
 
-    if (!bucket) {
-      return next(new errors.NotFoundError('Bucket not found'));
-    }
+      let bucketRemovalStorageEvents = results.map(function(result) {
+        return new StorageEvent(result);
+      });
 
-    bucket.remove(function(err) {
+      Bucket.findOne({
+      _id: req.params.id,
+      user: req.user._id
+      }, function(err, bucket) {
       if (err) {
         return next(new errors.InternalError(err.message));
       }
-      BucketEntry.remove({ bucket: req.params.id }, (err) => {
+
+      if (!bucket) {
+        return next(new errors.NotFoundError('Bucket not found'));
+      }
+
+      bucket.remove(function(err) {
         if (err) {
-          log.error('Unable to remove bucket entries, reason: %s',
-                    err.message);
+          return next(new errors.InternalError(err.message));
         }
+        BucketEntry.remove({ bucket: req.params.id }, (err) => {
+          if (err) {
+            log.error('Unable to remove bucket entries, reason: %s',
+                      err.message);
+          }
+          StorageEvent.collection.insert(bucketRemovalStorageEvents, function(err) {
```

Comment:
Local integration testing is throwing this warning, need to investigate. Most likely that collection.insert is not the appropriate method. 
0|bridge   | {"level":"info","message":"removing bucket f29c336153d99a9477f2d507 for patrick@storj.io","timestamp":"2017-03-22T01:07:27.107Z"}
0|bridge   | {"level":"warn","message":"Unable to save storage events, reason: Invalid Operation, No operations in bulk","timestamp":"2017-03-22T01:07:27.136Z"}